roles/confluent.kafka/tasks/main.yml

```diff
@@ -1,139 +1,139 @@
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "ssl"
   notify:
     - restart kafka
 - name: broker sasl_ssl config
   template:
     src: server_sasl_ssl.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "sasl_ssl"
   notify:
     - restart kafka
 - name: broker ssl custom signed config
   template:
     src: server_ssl_self_signed.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "ssl_customcerts"
   notify:
     - restart kafka
 - name: broker gssapi kerberos config
   template:
     src: server_kerberos.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "kerberos"
   notify:
     - restart kafka
 - name: broker ssl kerberos config
   template:
     src: server_kerberos_ssl.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "kerberos_ssl"
   notify:
     - restart kafka
 - name: broker ssl kerberos config
   template:
     src: server_kerberos_ssl.self_signed.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "kerberos_ssl_customcerts"
   notify:
     - restart kafka
 - name: broker Jaas config
   template:
     src: kafka_server_jaas.conf.j2
-    dest: "{{kafka.broker.jaas_file}}"
+    dest: "{{kafka.jaas_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "kerberos" or security_mode == "kerberos_ssl" or security_mode == "kerberos_ssl_customcerts"
   notify:
     - restart kafka
 - name: create systemd override file
   file:
-    path: "{{kafka.broker.systemd_override}}"
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    path: "{{kafka.systemd_override}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
```

Comment:
Recommend leaving in 'broker' as kafka is not a component but the platform name.